Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Routing.rst

```diff
@@ -729,10 +729,66 @@ two more options you can use:
 With ``suffix`` you can specify a custom filename suffix for the SubRoute. The ``variables`` option allows you to
 specify placeholders in the SubRoutes (see `Nested Subroutes`_).
 
+It also is possible to specify a `providerFactory` and (optional) `providerOptions` to generate the subroutes via the
```

Comment:
via are the `providerOptions` optional in this case? I think this is the general chapter which should explain how to use a custom `RoutesProviderFactory`?

Reply:
A custom provider might not need custom options and in this case is given an empty array, see https://github.com/neos/flow-development-collection/blob/ee8a8127e1d77908b5f44786a1ba4d6f2500c583/Neos.Flow/Classes/Mvc/Routing/ConfigurationRoutesProvider.php#L31

I think that totally makes sense and it's the way we do it for a lot of other similar configuration (patternOptions, postprocessorOptions, editorOptions, ...)

Reply:
Oke I thought so, will update the example so it’s not bound to the AttributeRoutes thing and more generic